src/app/component/user/components/profile/users-friends/friend-dashboard/all-friends/all-friends.component.ts

```diff
@@ -0,0 +1,127 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { MatSnackBarComponent } from '@global-errors/mat-snack-bar/mat-snack-bar.component';
+import { LocalStorageService } from '@global-service/localstorage/local-storage.service';
+import { FriendArrayModel, FriendModel, } from '@global-user/models/friend.model';
+import { UserFriendsService } from '@global-user/services/user-friends.service';
+import { Subject } from 'rxjs';
+import { catchError, takeUntil } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-all-friends',
+  templateUrl: './all-friends.component.html',
+  styleUrls: ['./all-friends.component.scss']
+})
+export class AllFriendsComponent implements OnInit, OnDestroy {
+
+  public userId: number;
+  public Friends: FriendModel[];
+  private destroy = new Subject();
+  public scroll: boolean;
+  public currentPage = 0;
+  public totalPages: number;
+
+  constructor(private userFriendsService: UserFriendsService,
+              private localStorageService: LocalStorageService,
+              private snackBar: MatSnackBarComponent) { }
+
+  ngOnInit() {
+    this.initUser();
+    this.getSixFriends();
+  }
+
+  public addStatus(userArray) {
+    userArray.forEach( elem => {
+      elem.added = true;
+    });
+  }
+
+  public changeStatus(id) {
+    const index = this.Friends.findIndex(elem => elem.id === id);
+    this.Friends[index].added = !this.Friends[index].added;
+  }
+
+  public getSixFriends() {
+    this.userFriendsService.getAllFriends(this.userId).pipe(
+      catchError((error) => {
+        this.snackBar.openSnackBar('error.message');
+        return error;
+      })
+    )
+    .pipe(
+      takeUntil(this.destroy)
+    )
+    .subscribe (
+      (data: FriendArrayModel) => {
+        this.Friends = data.page;
+        this.addStatus(this.Friends);
+      },
+    );
+  }
+
+  public onScroll(): void {
+    this.scroll = true;
+    this.currentPage += 1;
+    this.userFriendsService.getAllFriends(this.userId, this.currentPage).pipe(
+      catchError((error) => {
+        this.snackBar.openSnackBar('error.message');
+        return error;
+      })
+    )
+    .pipe(
+      takeUntil(this.destroy)
+    )
+    .subscribe(
+      (data: FriendArrayModel) => {
+        this.addStatus(data.page);
+        this.Friends = this.Friends.concat(data.page);
+      },
+     );
+  }
+
+  public deleteFriend(id: number) {
+    this.userFriendsService.deleteFriend(this.userId, id).pipe(
+      catchError((error) => {
+        this.snackBar.openSnackBar('error.message');
+        return error;
+      })
+    )
+    .pipe(
+      takeUntil(this.destroy)
+    )
```

Comment:
Some operators in a pipe are separated by comma: 
.pipe(
  operator1,
  operator2,
  ...
)

Reply:
I've fixed it.